__tests__/endpoints.test.js

```diff
@@ -142,4 +139,49 @@ describe('NC news endpoint tests',()=>{
             })
         })
     })
+    describe('POSTing news',()=>{
+        describe('POSTing new comment',()=>{
+            test('POST /api/articles/:article_id/comments', ()=>{
+                const commentToAdd = {username: "lurker", body: "this article is wack"}
+                const newComment = {
+                    body:  "this article is wack",
+                    votes: 0,
+                    author: "lurker",
+                    article_id: 13
+                  }
+                return request(app)
+                .post('/api/articles/13/comments')
+                .send(commentToAdd)
+                .expect(201)
+                .then(({body})=>{
+                    expect(body.comment.body).toBe(newComment.body);
+                    expect(body.comment.votes).toBe(newComment.votes);
+                    expect(body.comment.author).toBe(newComment.author);
+                    expect(body.comment.article_id).toBe(newComment.article_id);
+                    expect(typeof body.comment.created_at).toBe('string');
+                })
+            })
+            test('respond with 404 when provided a valid but non-existent article_id', ()=>{
+                const commentToAdd = {username: "lurker", body: "this article is wack"}
+                return request(app)
+                .post('/api/articles/9999/comments')
+                .send(commentToAdd)
+                .expect(404)
+                .then(({body})=>{
+                    expect(body.msg).toBe('article does not exist')
+                })
+            })
+            test('respond with 400 when provided an invalid param', ()=>{
+                const commentToAdd = {username: "lurker", body: "this article is wack"}
+                return request(app)
+                .post('/api/articles/meow/comments')
+                .send(commentToAdd)
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe('Bad Request')
+                })
+            })
+        })
+
```

Comment:
A few untested behaviours here:

- Status 201, ignores unnecessary properties: If the request body has any extra properties that are unnecessary, the post should still be successful with those properties ignored
- Status 400, missing required fields eg no username or body
- Status 404 username does not exist